src/triage/component/postmodeling/contrast/evaluator.py

```diff
@@ -0,0 +1,543 @@
+
+# Get individual model_ids from Audition outcome
+
+
+def get_models_ids(audited_model_group_ids):
+    query = db_engine.execute(text("""
+    SELECT model_group_id,
+           model_id
+    FROM model_metadata.models
+    WHERE model_group_id = ANY(:ids);
+    """), ids=audited_model_group_ids)
+
+    ModelTuple = namedtuple('Model', ['model_group_id', 'model_id'])
+    l = [ModelTuple(*i) for i in query]
+
+    return(l)
+
+# Get indivual model information/metadata from Audition output
+
+
+class Model(object):
+    '''
+    ModelExtractor class calls the model metadata from the database
+    and hold model_id metadata features on each of the class attibutes.
+    This class will contain any information about the model, and will be
+    used to make comparisons and calculations across models. 
+
+    A pair of (model_group_id, model_id) is needed to instate the class. These
+    can be feeded from the get models_ids. 
+    '''
+    def __init__(self, model_group_id, model_id):
+        self.model_id = model_id
+        self.model_group_id = model_group_id
+
+        # Retrive model_id metadata from the model_metadata schema
+        model_metadata = pd.read_sql("""
+        WITH
+        individual_model_ids_metadata AS(
+        SELECT m.model_id,
+               m.model_group_id,
+               m.hyperparameters,
+               m.model_hash,
+               m.experiment_hash,
+               m.train_end_time,
+               m.train_matrix_uuid,
+               m.training_label_timespan,
+               mg.model_type,
+               mg.model_config
+            FROM model_metadata.models m
+            JOIN model_metadata.model_groups mg
+            USING (model_group_id)
+            WHERE model_group_id = {}
+            AND model_id = {}
+        ),
+        individual_model_id_matrices AS(
+        SELECT DISTINCT ON (matrix_uuid)
+               model_id,
+               matrix_uuid
+            FROM test_results.predictions
+            WHERE model_id = ANY(
+                SELECT model_id
+                FROM individual_model_ids_metadata
+            )
+        )
+        SELECT metadata.*, test.*
+        FROM individual_model_ids_metadata AS metadata
+        LEFT JOIN individual_model_id_matrices AS test
+        USING(model_id);
+        """.format(self.model_group_id, self.model_id),
+                   con=db_engine)
+
+        # Add metadata attributes to model
+        self.model_type = model_metadata.loc[0, 'model_type']
+        self.hyperparameters = model_metadata.loc[0, 'hyperparameters']
+        self.model_hash = model_metadata.loc[0, 'model_hash']
+        self.train_matrix_uuid = model_metadata.loc[0, 'train_matrix_uuid']
+        self.pred_matrix_uuid = model_metadata.loc[0, 'matrix_uuid']
+        self.train_matrix = None
+        self.pred_matrix = None
+        self.preds = None
+        self.feature_importances = None
+        self.model_metrics = None
+
+    def __str__(self):
```

Comment:
Use `__repr__` instead of `__str__` (see https://stackoverflow.com/a/2626364/754176)